electrum/lnrouter.py

```diff
@@ -161,12 +167,251 @@ def is_fee_sane(fee_msat: int, *, payment_amount_msat: int) -> bool:
+            self._cannot_send_forward = None
+
+    @property
+    def can_send_backward(self):
+        return self._can_send_backward
+
+    @can_send_backward.setter
+    def can_send_backward(self, amount):
+        if self._can_send_backward and self._can_send_backward > amount:
+            return
+        self._can_send_backward = amount
+        if self._cannot_send_backward and self._can_send_backward > self._cannot_send_backward:
+            self._cannot_send_backward = None
+
+    @property
+    def cannot_send_forward(self):
+        return self._cannot_send_forward
+
+    @cannot_send_forward.setter
+    def cannot_send_forward(self, amount):
+        # we don't want to record less significant info
+        # (not sendable amount is higher than known not sendable amount):
+        if self._cannot_send_forward and self._cannot_send_forward < amount:
+            return
+        self._cannot_send_forward = amount
+        if self._can_send_forward and self._can_send_forward > self._cannot_send_forward:
+            self._can_send_forward = None
+        # if we can't send over the channel, we should be able to send in the
+        # reverse direction
+        self.can_send_backward = amount
+
+    @property
+    def cannot_send_backward(self):
+        return self._cannot_send_backward
+
+    @cannot_send_backward.setter
+    def cannot_send_backward(self, amount):
+        if self._cannot_send_backward and self._cannot_send_backward < amount:
+            return
+        self._cannot_send_backward = amount
+        if self._can_send_backward and self._can_send_backward > self._cannot_send_backward:
+            self._can_send_backward = None
+        self.can_send_forward = amount
+
+    def can_send(self, is_forward_direction: bool):
+        # make info invalid after some time?
+        if is_forward_direction:
+            return self.can_send_forward
+        else:
+            return self.can_send_backward
+
+    def cannot_send(self, is_forward_direction: bool):
+        # make info invalid after some time?
+        if is_forward_direction:
+            return self.cannot_send_forward
+        else:
+            return self.cannot_send_backward
+
+    def update_can_send(self, is_forward_direction: bool, amount: int):
+        if is_forward_direction:
+            self.can_send_forward = amount
+        else:
+            self.can_send_backward = amount
+
+    def update_cannot_send(self, is_forward_direction: bool, amount: int):
+        if is_forward_direction:
+            self.cannot_send_forward = amount
+        else:
+            self.cannot_send_backward = amount
+
+    def __repr__(self):
+        return f"forward: can send: {self._can_send_forward}, cannot send: {self._cannot_send_forward}, \n" \
+               f"backward: can send: {self._can_send_backward} cannot send: {self._cannot_send_backward}, \n" \
+               f"blacklisted: {self.blacklisted}"
+
+
+class LiquidityHints:
```

Comment:
Please no names where one name is X and the other is X in plural.
Maybe `LiquidityHintMgr` instead.

Reply:
Done